### PR TITLE
Fjc robots

### PR DESCRIPTION
**Pull request by andriioleshko**

### CODE REVIEW

File: src/robots/AdvancedRobot.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package robots;
```
Comment (Maks-Chornyi) [inline src/robots/AdvancedRobot.java]: first of all, please, follow the correct java project structure. it has to be src/main/java and all packeges inside. so src/main/java/robots/and your claseses (or other pachages) are here

File: src/robots/Application.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package robots;
+
+import java.util.Random;
+
+public class Application {
```
Comment (Maks-Chornyi) [inline src/robots/Application.java]: look at your project's structure. It has application class (main class). robots' classes. and some files (interfaces) related to robot's actions. so in this case you can add additional packages. for instance you can create package robot and put there all robot's classes (primitive robo, advanced robot), also you can create package 'action' and put there your 'actions' realted to robot. movement, shoot and so on, in this case your program will look much better and would be easier for understanding for others.

File: src/robots/Application.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    public static void main(String[] args) {
+        Random rand = new Random();
+        int randomInt = rand.nextInt(12)+1;
```
Comment (Maks-Chornyi) [inline src/robots/Application.java]: right click on the class, reformate code. please don't forget to do this, as it makes your code cleaner. (in this case you forgot to put spaces)

File: src/robots/Defence.java [hunk 4]
```diff
@@ -0,0 +1,3 @@
+package robots;
+
+public interface Defence {String defence();}
```
Comment (Maks-Chornyi) [inline src/robots/Defence.java]: the correct (conventional) structure of the method is:
signature of the method and first curly brace is in one line. than goes 'new line'. in the new line is method's body (if exists, in our case it is not, as this is interface). 
when body ends, new line and curly brace which closes the method's body and ends it. so no just one line. but like this:
public interface Defence {
____String defence();
}
(three lines should be here)

File: src/robots/Move.java [hunk 5]
```diff
@@ -0,0 +1,3 @@
+package robots;
+
+public interface Move {String move();}
```
Comment (Maks-Chornyi) [inline src/robots/Move.java]: same as above

File: src/robots/PrimitiveRobot.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package robots;
+
+public class PrimitiveRobot extends Robot implements Yell, Move, Shoot {
+    public PrimitiveRobot(String name) {super(name, "1.0.0");}
+
+    @Override
+    public String startYelling() {return "I can't yell at you, Master.";}
```
Comment (Maks-Chornyi) [inline src/robots/PrimitiveRobot.java]: same as above

File: src/robots/PrimitiveRobot.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+    @Override
+    public String move() {return "I cannot move.";}
```
Comment (Maks-Chornyi) [inline src/robots/PrimitiveRobot.java]: +1

File: src/robots/PrimitiveRobot.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+
+    @Override
+    public String shoot() {return "I'll try my best to shoot.";}
```
Comment (Maks-Chornyi) [inline src/robots/PrimitiveRobot.java]: +1

File: src/robots/Robot.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    private String version;
+
+    public Robot(String name, String version) {
+        this.name = name;
+        this.version = version;
+    }
+
+    public String getName() {
+        return name;
+    }
```
Comment (Maks-Chornyi) [inline src/robots/Robot.java]: this method looks good (method's body in the new line, good)

File: src/robots/Robot.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    @Override
+    public String startYelling() {
+        return null;
+    }
+
+    @Override
+    public String defence() {
+        return "Nobody can't override my functions and force to do what is banned for me when I was created :).";
+    }
+}
```
Comment (Maks-Chornyi) [inline src/robots/Robot.java]: add new line to the end of the file

File: src/robots/Shoot.java [hunk 11]
```diff
@@ -0,0 +1,3 @@
+package robots;
+
+public interface Shoot {String shoot();}
```
Comment (Maks-Chornyi) [inline src/robots/Shoot.java]: +1

File: src/robots/StandartRobot.java [hunk 12]
```diff
@@ -0,0 +1,14 @@
+package robots;
+
+public class StandartRobot extends Robot implements Yell, Move, Shoot {
+    public StandartRobot(String name) {super(name, "1.0.3"); }
+
+    @Override
+    public String startYelling() {return "Hey!";}
```
Comment (Maks-Chornyi) [inline src/robots/StandartRobot.java]: same

File: src/robots/Robot.java [hunk 13]
```diff
@@ -0,0 +1,39 @@
+package robots;
+
+public class Robot implements Shoot, Move, Yell, Defence {
```
Comment (Maks-Chornyi) [inline src/robots/Robot.java]: your Robor class looks really like abstract class. it just contain 'common' logic/fields. make it abstract. but also read about the exact definition of 'abstract class' and add corresponding method(s).

File: src/robots/Robot.java [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+    private String name;
+    private String version;
```
Comment (Maks-Chornyi) [inline src/robots/Robot.java]: you craeted version field, but it is string. well, it's ok, but I want you to use Enum. please read/watch about enums in java and change this field from String to Enum and provide different enum for each version.

**Review by Maks-Chornyi — COMMENTED** [review]
left few comments, check them out, please :)

File: src/robots/AdvancedRobot.java (hunk 1 above)
Comment (andriioleshko) [reply src/robots/AdvancedRobot.java]: restructured

File: src/robots/Application.java (hunk 2 above)
Comment (andriioleshko) [reply src/robots/Application.java]: restructured

File: src/robots/Application.java (hunk 3 above)
Comment (andriioleshko) [reply src/robots/Application.java]: reformated code

File: src/robots/Defence.java (hunk 4 above)
Comment (andriioleshko) [reply src/robots/Defence.java]: reformated code

File: src/robots/Move.java (hunk 5 above)
Comment (andriioleshko) [reply src/robots/Move.java]: reformated code

File: src/robots/PrimitiveRobot.java (hunk 6 above)
Comment (andriioleshko) [reply src/robots/PrimitiveRobot.java]: reformated code

File: src/robots/PrimitiveRobot.java (hunk 7 above)
Comment (andriioleshko) [reply src/robots/PrimitiveRobot.java]: reformated code

File: src/robots/PrimitiveRobot.java (hunk 8 above)
Comment (andriioleshko) [reply src/robots/PrimitiveRobot.java]: reformated code

File: src/robots/Robot.java (hunk 10 above)
Comment (andriioleshko) [reply src/robots/Robot.java]: added new line

File: src/robots/Shoot.java (hunk 11 above)
Comment (andriioleshko) [reply src/robots/Shoot.java]: reformated code

File: src/robots/StandartRobot.java (hunk 12 above)
Comment (andriioleshko) [reply src/robots/StandartRobot.java]: reformated code

File: src/robots/Robot.java (hunk 13 above)
Comment (andriioleshko) [reply src/robots/Robot.java]: made class abstract, modified some methods to abstract (removed the bodies from there)

File: src/robots/Robot.java (hunk 14 above)
Comment (andriioleshko) [reply src/robots/Robot.java]: added enum, used in abstract class

**Review by Maks-Chornyi — APPROVED** [review]
Looks good, nice.

**Pull request merged**